### PR TITLE
Install keras, tensorflow and check if accessible in Dockerfile

### PR DESCRIPTION
**Pull request by cdeepakroy**
This PR:

* Adds keras and tensorflow to requirements.txt. Note that only CPU version of tensorflow is in requirements.txt. If the user wants to take advantage of the GPU, then the GPU version (`tensorflow-gpu`), the user is responsible for uninstalling the CPU version and installing the GPU version explicitly after installing HistomicsTK. This is clarified in the documentation for installing HistomicsTK as a python toolkit.

* Creates a separate virtual environment called `venv-gpu` and installs `tensorflow-gpu` in the Dockerfile with HistomicsTK CLIs.

* Adds a new shell script called `server/docker-entrypoint.sh` that activates the `venv-gpu` environment only if the docker container is run with `nvidia-docker` and if `nvidia-smi` works. This is to ensure that the CLIs run on all machines (with or without GPU) irrespective of whether they are run with plain `docker run` or `nvidia-docker run`.

Note that the CLIs in the docker container will take advantage of the GPU only if run with `nvidia-docker` that is expected to be pre-installed on the machine.

### CODE REVIEW

File: server/docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+if type nvidia-smi > /dev/null;
```
Comment (manthey) [inline server/docker-entrypoint.sh]: Can you change this to
```
if type nvidia-smi >/dev/null 2>/dev/null;
```
to also suppress the `type: nvidia-smi: not found` error message?
